crates/sui-graphql-rpc/src/types/transaction_block_kind/programmable.rs

```diff
@@ -0,0 +1,486 @@
+    SharedInput(SharedInput),
+    Receiving(Receiving),
+    Pure(Pure),
+}
+
+/// A Move object, either immutable, or owned mutable.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+#[graphql(complex)]
+struct OwnedOrImmutable {
+    address: SuiAddress,
+    version: u64,
+    digest: String,
+}
+
+/// A Move object that's shared.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct SharedInput {
+    address: SuiAddress,
+    /// The version that this this object was shared at.
+    initial_shared_version: u64,
+    /// Controls whether the transaction block can reference the shared object as a mutable
+    /// reference or by value.
+    mutable: bool,
+}
+
+/// A Move object that can be received in this transaction.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+#[graphql(complex)]
+struct Receiving {
+    address: SuiAddress,
+    version: u64,
+    digest: String,
+}
+
+/// BCS encoded primitive value (not an object or Move struct).
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct Pure {
+    /// BCS serialized and Base64 encoded primitive value.
+    bytes: Base64,
+}
+
+/// A single transaction, or command, in the programmable transaction block.
+#[derive(Union, Clone, Eq, PartialEq)]
+enum ProgrammableTransaction {
+    MoveCall(MoveCallTransaction),
+    TransferObjects(TransferObjectsTransaction),
+    SplitCoins(SplitCoinsTransaction),
+    MergeCoins(MergeCoinsTransaction),
+    Publish(PublishTransaction),
+    Upgrade(UpgradeTransaction),
+    MakeMoveVec(MakeMoveVecTransaction),
+}
+
+#[derive(Clone, Eq, PartialEq)]
+struct MoveCallTransaction(NativeMoveCallTransaction);
+
+/// Transfers `inputs` to `address`. All inputs must have the `store` ability (allows public
+/// transfer) and must not be previously immutable or shared.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct TransferObjectsTransaction {
+    /// The objects to transfer.
+    inputs: Vec<TransactionArgument>,
+
+    /// The address to transfer to.
+    address: TransactionArgument,
+}
+
+/// Splits off coins with denominations in `amounts` from `coin`, returning multiple results (as
+/// many as there are amounts.)
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct SplitCoinsTransaction {
+    /// The coin to split.
+    coin: TransactionArgument,
+
+    /// The denominations to split off from the coin.
+    amounts: Vec<TransactionArgument>,
+}
+
+/// Merges `coins` into the first `coin` (produces no results).
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct MergeCoinsTransaction {
+    /// The coin to merge into.
+    coin: TransactionArgument,
+
+    /// The coins to be merged.
+    coins: Vec<TransactionArgument>,
+}
+
+/// Publishes a Move Package.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct PublishTransaction {
+    /// Bytecode for the modules to be published, BCS serialized and Base64 encoded.
+    modules: Vec<Base64>,
+
+    /// IDs of the transitive dependencies of the package to be published.
```

Comment:
Curious about this `dependencies` field: can I safely assume that these dependencies are move objects?

Reply:
I see, did not think about a failing transaction.